### ISSUE TITLE
Basic needs for this repo to be useful

### ISSUE THREAD

**Issue by jeremyeder** [kubernetes/perf-tests#5]
At least:
- Which sig owns this repository
- An architecture discussion
- An agreed-upon automation framework
- Modularized to support also running on distributions of Kubernetes
- Modularized so there is no dependency on any cloud/infra
- Some understanding of how the merge queue works...don't want delays.
- Discussion around what tests to move from e2e
- CI of the tests within the repo
- Policy of unit tests for the tests themselves
- CI of running the tests themselves
- Does this repository depend on kubernetes e2e tests
  - Can this repository import useful framework helpers/functions from e2e

**Comment by zilman** [comment]
## (CLI) Conventions 

> Because in general tools are independent and have their own ways of being configured or run

Perhaps it doesn't have to be that way. I'd like to suggest the following;

- Python 
- [Click](http://click.pocoo.org/5/) library 
- Yaml

It looks like we'll have several projects contributing stuff and a lot of people will want to expose running a test in a nice way with a cli and maybe even as a subcommand of their project. I know I want that.

Aforementioned Click lets you compose these things pretty easily. For example, on @jeremyeder's side of things they have [this runner](https://github.com/sjug/svt/blob/531aad992ed6c7fa2772627b98b153994171285f/openshift_scalability/cluster-loader.py) that I could simply import as a subcommand if it were to adopt the convention. 

Exposing it in such a way is more user friendly than having users pull this repo and individually read and understand each subdirectory. 

Furthermore I think each cli should be able to parse a yaml config file (just like kubectl) that defines a test. See this [webframework comparison benchmark](https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/Python/falcon/benchmark_config.json) for inspiration. 

### In Summary

I should be able to pull this repo, cd into the subdirectory of interest, and have a cli.py wrapper I can immediately try out either against example yaml definitions or with my own flags. 

I should also easily be able to import into my own project whichever subdirectories of this repo I find relevant. 

Thoughts?

**Comment by jeremyeder** [comment]
Meeting occurred, notes were taken:
https://docs.google.com/document/d/1x1r65mXbX_RgLA0HTlsb8QRQqW0xB5XSM1vrMb2naQo/edit

**Comment by gmarek** [comment]
@zilman - there will be very different things here, e.g. running kubemark will be drastically different than running cluster-loader or dedicated component benchmarks and not all of them will have it's own CLI. Kubemark uses `kubectl`, dedicated benchmarks and e2e tests will probably use framework from the main repo, etc. OTOH cluster loader probably should have a CLI, but I don't think it will be shared with anything else, so we'll go with whatever is easiest.

**Comment by pskrzyns** [comment]
For the existing e2e tests(density, load) i.e. density test: ```should.allow.starting.100.pods.per.node``` it would be nice to switch them to: ```should.allow.starting.10.pods.per.core``` and during the test calculate how many pods should be running on the single node. At this moment biggest density test is 100 pods per node so in case of running it on bare metal(depending on hardware configuration) it could give only about 2 pods per core.

**Comment by jeremyeder** [comment]
@sjug is going to discuss the first part of this at the @kubernetes/sig-testing tomorrow, Tue Oct 4.  And then in further depth at the @kubernetes/sig-scalability meeting on Thursday Oct 6, which we've agreed will own the "perf-tests" repo.

**Comment by gmarek** [comment]
@pskrzyns - I don't think 10pods/core are already supported by Kubelet (@yujuhong), plus this issue is about moving tests to separate repo, not updating them:)

**Comment by yujuhong** [comment]
> @pskrzyns - I don't think 10pods/core are already supported by Kubelet (@yujuhong), plus this issue is about moving tests to separate repo, not updating them:)

Agreed. For the pods per core discussion, please move it to the original issue: https://github.com/kubernetes/kubernetes/issues/25762
